include/mbedtls/config.h

```diff
@@ -1706,6 +1706,25 @@
  */
 #define MBEDTLS_SSL_PROTO_TLS1_2
 
+/**
+ * \def MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL
```

Comment:
@yanesca I'd rather not try to track in detail what has been implemented and what not within the source files, since it complicates matters and is of no benefit to the user IMO insofar as it is about features that aren't yet accessible from the public API.

Reply:
I didn't mean it for the sake of the user, I would like to have that documentation for the maintainer team's sake. The goal of this documentation would be to make it easier to assess where we are, get up to speed if someone new starts reviewing the next bit, or just if there is a long time between reviews.

Again, I wouldn't want to ask for full and detailed architecture description and I don't think that is worth going into excessive details or giving instructions and documentation for hypothetical users.

Reply:
@yanesca I'll add a short document where we can keep track of what's already implemented under the umbrella of `MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL`.